Added stop-bits field to uart-controller.yaml,

avigreen1978 opened the pull request:
using in drivers/serial/uart_stm32.c .
For some reason, stop-bits option was not configurable
from dts (was hard-coded).
This means that if another stop-bits configuration is desired,
you have to call uart_configure on runtime.
There's no reason for that.

Signed-off-by: Avi Green <avigreen1978@yandex.com>

dcpleung reviewed (changes requested):
Please split this into two commits: one for the DTS bindings, and another for the STM32 driver change.

erwango commented on dts/bindings/serial/uart-controller.yaml:
```diff
@@ -29,3 +29,12 @@ properties:
+    stop-bits:
+      required: false
+      type: string
+      description: Inital stop-bits setting for UART
```
Please add a comment about default value.

henrikbrixandersen replied:
Adding this to the generic uart-controller binding but only support to the STM32 UART driver will likely lead to user confusion.

erwango replied:
I guess that all drivers have a default value for this, isn't it ?

henrikbrixandersen replied:
Sure, but adding a generic devicetree property for this gives the impression that all drivers supports that property. This is not the case with this PR, since it is only adding support for it to the STM32 UART driver.

erwango replied:
ok, so your advice is:
- Extend this to other serial drivers
or
- Move to STM32 binding 

Correct ?

henrikbrixandersen replied:
@erwango Correct.

avigreen1978 replied:
I can move it to stm32 binding, but this might make other users not aware of this feature.
On the other hand, I don't think it's correct to ask me to extend this to other drivers since I don't know the implementation details of them. (I only work with stm32, I don't know the 
Which option do you prefer?

mbolivar-nordic replied:
> Let's move to STM32 binding for now if you're not comfortable with extending the change on other drivers

I agree.

mbolivar-nordic commented on dts/bindings/serial/st,stm32-usart.yaml:
```diff
@@ -20,3 +20,13 @@ properties:
+        - "1"
+        - "0.5"
+        - "1.5"
+        - "2"
```
Please add the missing newline to end of file.

avigreen1978 committed:
dts-bindings: uart-controller - Added stop-bits field to st,stm32-usart

For some reason, stop-bits option was not configurable
from dts (was hard-coded).
This means that if another stop-bits configuration is desired,
you have to call uart_configure on runtime.
There's no reason for that

Signed-off-by: Avi Green <avigreen1978@yandex.com>

avigreen1978 committed:
serial: uart_stm32: Use stop-bits dts field

uart_stm32.c driver now takes stop-bits field from dts ,
configures it in init function

Signed-off-by: Avi Green <avigreen1978@yandex.com>

erwango commented on dts/bindings/serial/st,stm32-usart.yaml:
```diff
@@ -20,3 +20,13 @@ properties:
         For example the USART1 would be
            pinctrl-0 = <&usart1_tx_pb6 &usart1_rx_pb7>;
+
+    stop-bits:
```
Please also add this to `st,stm32-uart.yaml` and `st,stm32-lpuart.yaml` and we'll be fine

erwango commented on dts/bindings/serial/st,stm32-usart.yaml:
```diff
@@ -20,3 +20,13 @@ properties:
+    stop-bits:
+      required: false
+      type: string
+      description: Inital stop-bits setting for UART
```
Nitpick: Let's replace "Initial" by "Boot time"

FRASTM replied:
"Init**i**al"

erwango reviewed (changes requested):
Very last changes. Thanks for your patience.

dleach02 commented:
@avigreen1978 do you plan on revisiting this PR? There are conflicts that need to be resolved and you need to work with @erwango to resolve his block

erwango commented:
> @avigreen1978 do you plan on revisiting this PR? There are conflicts that need to be resolved and you need to work with @erwango to resolve his block

I guess a rebase in a first time would help.
@avigreen1978 would you have time to have a look ?

FRASTM commented on drivers/serial/uart_stm32.c:
```diff
@@ -1463,12 +1464,12 @@ static int uart_stm32_init(const struct device *dev)
 	}
-
+	ll_stopbits = uart_stm32_cfg2ll_stopbits(config->stop_bits);
 	/* Set datawidth and parity, 1 start bit, 1 stop bit  */
```
could the `uart_stm32_cfg2ll_stopbits()`  raise an error if configuring a non-supported stop-bit
For example DTS defines 0.5 but the soc/uart does not support

henrikbrixandersen commented:
This needs another rebase.
